libs/core/gasnet_base/include/hpx/gasnet_base/gasnet.hpp

```diff
@@ -0,0 +1,32 @@
+//  Copyright (c) 2017 Mikael Simberg
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#pragma once
+
+#if (defined(HPX_HAVE_NETWORKING) && defined(HPX_HAVE_PARCELPORT_GASNET)) ||   \
+    defined(HPX_HAVE_MODULE_GASNET_BASE)
+
+#if defined(__clang__)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wcast-qual"
+#elif defined(__GNUC__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wcast-qual"
+#endif
+
+#ifndef GASNET_PAR
+#define GASNET_PAR 1
+#endif
```

Comment:
I don't understand the pkg-config magic being used in cmake/HPX_SetupGasnet.cmake.

However if the three lines above cannot be safely deleted that tells me the build infrastructure is not correctly using `pkg-config` to pick up the GASNet compile flags required by `gasnet.h`. This define happens to be unconditionally required everywhere, but other defines provided in those same flags may be conduit-specific or system-specific, such that things might appear to work for one conduit/system and then break on another. IOW hard-coding this define here is fragile and may hide more subtle build problems.

Please see the "Using GASNet with pkg-config" section in [the GASNet README](https://gasnet.lbl.gov/dist-ex/README) for the supported/documented way to build against the GASNet headers. In particular, the following are probably missing from the flags used to compile this file:

* `pkg-config gasnet-$(conduit)-par --variable=GASNET_CXXCPPFLAGS`  - preprocessor options
* `pkg-config gasnet-$(conduit)-par --variable=GASNET_CXXFLAGS` - compiler options

Omitting either may lead to mysterious build errors or even miscompilation of the header on some platforms.
Other documented variables contain subsets of these, for example `GASNET_INCLUDES` if you need just the `-I` options or `GASNET_DEFINES` for just the `-D` options.

Reply:
@bonachea nice catch. I can remove those 3 lines and everything should be good. The cmake code corrects for some incompatibilities between cmake and pkgconfig.